web/blog/2021-03-05-postgres-as-a-cron-server.md

```diff
@@ -0,0 +1,191 @@
+
+It is. Here's how.
+
+
+<!--truncate-->
+
+### What's cron?
+
+A "cron job" is a script[^1] that runs periodically at fixed times, dates, or intervals. Traditionally you'd set it up on a Linux server. An example might be an hourly script that downloads emails to your computer. 
+
+[^1]: Not necessarily a script. The cron is really a scheduler which triggers a job (of some sort, usually a bash script).
+
+These days, cron jobs are set up on a remote servers and in the cloud to run internet-related tasks. Like checking an endpoint every hour, or scraping a website every day.
+
+### Postgres + cron
+
+Postgres has "extensions" which allow you to, well, extend the database with "non-core" features. Extensions essentially turn Postgres into an application server.
+
+The team at [Citus](https://github.com/citusdata) created [`pg_cron`](https://github.com/citusdata/pg_cron) to run periodic jobs within your Postgres database. 
+
+
+#### Enabling the extension
+
+If you're using a cloud-hosted Postgres database, make sure that `pg_cron` is installed first. The easiest way to do this is to run this command:
+
+```sql
+select name, comment, default_version, installed_version
+from pg_available_extensions
+where name = 'pg_cron';
+```
+
+If it returns a result then the extension is supported and you can turn it on by running:
+
+
+```sql
+create extension if not exists pg_cron;
+```
+
+If you're using Supabase you can also enable it in the Dashboard.
+
+
+![This image shows that pg_cron is enabled in the Supabase Dashboard](/img/blog/supabase-extensions.png)
+
+
+
+### Postgres + webhooks
+
+The Supabase customer wanted to call external endpoints every day. How would we do this? Another extension of course. This time we're going to use [pgsql-http](https://github.com/pramsey/pgsql-http) by [@pramsey](https://github.com/pramsey). Using the same technique, we can enable the extension (if it exists in your cloud provider).
+
+```sql
+create extension if not exists http;
+```
+
+This extension can now be used for [sending](https://github.com/pramsey/pgsql-http#functions) `GET`, `POST`, `PATCH`, and `DELETE` requests.
+
+For example, this function would get all the people in Star Wars (using the [Star Wars API](https://swapi.dev)):
+
+```sql
+select content::json->'results' 
+from http_get('https://swapi.dev/api/people');
+```
+
+
+### Postgres + cron + webhooks
+
+Now the fun stuff. For this example we're going to call [webhook.site](https://webhook.site) every minute with the payload `{ "hello": "world" }`. 
+
+Here's the code (with comments `--like this`).
+
+```sql
+select
+  cron.schedule(
+    'webhook-every-minute', -- name of the cron job
+    '* * * * *', -- every minute
+    $$ 
+    select status
+    from
+      http_post(
+        'https://webhook.site/223c8a43-725b-4cbd-b1fe-d0da73353a6b', -- webhook URL
```

Comment:
```suggestion
        'https://webhook.site/223c8a43-725b-4cbd-b1fe-d0da73353a6b', -- webhook URL, replace the ID(223c8..) with your own
```

Reply:
The example url gives 404 otherwise: https://webhook.site/223c8a43-725b-4cbd-b1fe-d0da73353a6b. Better to be explicit about this.